src/fsharp/fsc.fs

```diff
@@ -1201,11 +1498,11 @@ module StaticLinker =
                    (mkILMethods (topTypeDefs |> List.collect (fun td -> td.Methods.AsList)), 
                     mkILFields (topTypeDefs |> List.collect (fun td -> td.Fields.AsList)))
 
-            let ilxMainModule = 
-                { ilxMainModule with 
+            let ilxMainModule =
+                { ilxMainModule with
                     Manifest = (let m = ilxMainModule.ManifestOfAssembly in Some {m with CustomAttrsStored = storeILCustomAttrs (mkILCustomAttrs (m.CustomAttrs.AsList @ savedManifestAttrs)) })
                     CustomAttrsStored = storeILCustomAttrs (mkILCustomAttrs [ for m in moduls do yield! m.CustomAttrs.AsArray ])
-                    TypeDefs = mkILTypeDefs (topTypeDef :: List.concat normalTypeDefs)
+                    TypeDefs = typeForwarding.TypeForwardILTypeDefs(topTypeDef :: List.concat normalTypeDefs)
```

Comment:
Can the 400 lines of code from line 1175 be replaced by 

    let ilxMainModule2 = morphILTypeRefsInILModuleMemoized typeForwardILTypeRef ilxMainModule 

thanks

Reply:
@dsyme, done … I wish I had discovered that function a week ago.